project/src/pages/FilmDetailsPage/index.tsx

```diff
@@ -1,25 +1,37 @@
 import { Helmet } from 'react-helmet-async';
+import { useEffect } from 'react';
 import { Link, useParams } from 'react-router-dom';
 import { Footer } from '../../components/Footer/index';
-import { films } from '../../mock-server/films';
+import { useAppDispatch, useAppSelector} from '../../hooks';
 import { REGEX_ALT } from '../../constants';
 import { createAltText } from '../../utils/createAltText';
 import { FilmCard } from '../../components/FilmCard/index';
 import { DetailsNavigation } from '../../components/DetailsNavigation/index';
-import { findSimilarFilms } from './helper';
 import { Logo } from '../../components/Logo';
+import {fetchFilmAction, fetchSimilarFilmsAction} from '../../store/api-action';
+
 
 export const FilmDetailsPage = () => {
+  const dispatch = useAppDispatch();
   const { id } = useParams<string>();
-  const film = films.find((item) => item.id === id);
+
+  useEffect(() => {
```

Comment:
Let's fix unnesseccary requests  in all useEffect hooks
https://redux-toolkit.js.org/api/createAsyncThunk#canceling-while-running

Reply:
fixed